runtime/fixtures/dart_tool/flutter_build/dart_plugin_registrant.dart

```diff
@@ -31,3 +34,38 @@ void mainForPluginRegistrantTest() {
 }
 
 void main() {}
+
+void dartPluginRegistrantIsolate(SendPort sendPort) {
+  DartPluginRegistrant.ensureInitialized();
+  sendPort.send(didCallRegistrantBeforeEntrypoint);
+}
+
+@pragma('vm:entry-point')
+void callDartPluginRegistrantFromBackgroundIsolate() async {
+  ReceivePort receivePort = ReceivePort();
+  Isolate isolate = await Isolate.spawn(dartPluginRegistrantIsolate, receivePort.sendPort);
```

Comment:
It would be great if the Dart C++ API supported registering a callback that triggers whenever a new isolate is spawn. 

The main benefit is that it would keep `DartPluginRegistrant.ensureInitialized()` as implementation detail.

Reply:
We want people to have the option of not calling the dart plugin registrant on background isolates though for performance reasons.  If you are spawning an isolate to do background encryption you don't want to start executing whatever happens in whatever plugins you are using elsewhere in your app.

Reply:
right. Although, the registration phase in federated plugins is about binding to a specific implementation of the shared interface, so the cost might just be constructing that specific implementation object for the plugins that do have a Dart plugin registrant.